internal/processor/parser/app.go

```diff
@@ -32,27 +26,25 @@ func newAppCache() appCache {
 }
 
 // Add inserts or update a new app in the cache, and returns it
-func (c *appCache) Add(cfApp cfclient.App) *App {
+func (c *appCache) Add(cfApp cloudfoundry.CFApplication) (*App, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	if app := c.apps[cfApp.Guid]; app != nil {
-		app.setAppData(cfApp)
+	if app := c.apps[cfApp.GUID]; app != nil {
+		err := app.setAppData(cfApp)
+		if err != nil {
+			return nil, err
+		}
 	} else {
-		app := newApp(cfApp.Guid)
-		app.setAppData(cfApp)
-		c.apps[cfApp.Guid] = app
+		app := newApp(cfApp.GUID)
+		err := app.setAppData(cfApp)
+		if err != nil {
+			return nil, err
+		}
+		c.apps[cfApp.GUID] = app
 	}
 
-	return c.apps[cfApp.Guid]
-}
-
-// Delete removes an app from the cache
-func (c *appCache) Delete(guid string) {
```

Comment:
Are we never deleting apps from the cache ?

Reply:
Seems like we're not. Do you think this could be a problem? IMHO this would be addressed by nozzle instance restarting (thus doing a new cache warmup), which I guess would happen from time to time. Are the nozzle instances so long-lived that this could make us run into memory issues?

Reply:
I'm not entirely sure. It's just that I thought we had some logic that would remove apps from the cache when trying to fetch them and it resulted in errors. But I guess I removed that part when first refactoring the cache and I forgot.
You're right I don't think app churn is something that would be an issue and cause memory issues